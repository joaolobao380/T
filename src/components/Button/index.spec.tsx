import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';

import { Button } from '.';

describe('LoadingButton', () => {
  it('Should display the title', () => {
    const { getByText } = render(<Button title="Click Me" isLoading={false} onPress={() => {}} />);
    expect(getByText('Click Me')).toBeTruthy();
  });

  it('Should show an activity indicator when loading', () => {
    const { getByTestId } = render(<Button title="Loading" isLoading onPress={() => {}} />);
    expect(getByTestId('activity-indicator')).toBeTruthy();
  });

  it('should call onPress when pressed', () => {
    const mockOnPress = jest.fn();
    const { getByText } = render(
      <Button title="Press Me" isLoading={false} onPress={mockOnPress} />
    );

    const button = getByText('Press Me');
    fireEvent.press(button);
    expect(mockOnPress).toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const { toJSON } = render(<Button title="Click Me" isLoading={false} onPress={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const { toJSON } = render(<Button title="Loading..." isLoading onPress={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
