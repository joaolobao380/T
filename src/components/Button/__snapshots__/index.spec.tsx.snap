// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton Renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#2A2A2A",
        "borderRadius": 4,
        "justifyContent": "center",
        "paddingVertical": 16,
        "width": "100%",
      },
      false,
      false,
    ]
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Poppins-Regular",
        "fontSize": 16,
      }
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`LoadingButton Renders correctly when loading 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#2A2A2A",
        "borderRadius": 4,
        "justifyContent": "center",
        "paddingVertical": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "#444444",
      },
      false,
    ]
  }
>
  <ActivityIndicator
    color="#FFFFFF"
    size="small"
    testID="activity-indicator"
  />
</View>
`;
